src/algebra/lie_algebras.lean

```diff
@@ -0,0 +1,138 @@
+/-
+Copyright (c) 2019 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import ring_theory.algebra data.matrix.basic
+
+/-!
+# Lie algebras
+
+This file defines Lie rings, and Lie algebras over a commutative ring. It shows how these arise from
+associative rings and algebras via the ring commutator. In particular it defines the Lie algebra
+of endomorphisms of a module as well as of the algebra of square matrices over a commutative ring.
+
+## Notations
+
+We introduce the notation ⦋x, y⦌ for the Lie bracket. Note that these are the Unicode "square
+brackets with underbar" rather than the usual square brackets.
+
+## Implementation notes
+
+Lie algebras are defined as modules with a compatible Lie ring structure, and thus are partially
+unbundled. Since they extend Lie rings, these are also partially unbundled.
+
+## References
+* [N. Bourbaki, *Lie Groups and Lie Algebras, Chapters 1--3*][bourbaki1975]
+
+## Tags
+
+lie ring, lie algebra, lie bracket, jacobi identity, ring commutator
+-/
+
+universes u v
+
+class has_bracket (L : Type v) := (bracket : L → L → L)
+
+notation `⦋`x`,` y`⦌` := has_bracket.bracket x y
```

Comment:
Should it be a localized notation?

Reply:
Apparently these aren't very common unicode characters... my browser font doesn't know how to display them...
I could update my browser font, but it still means that lot's of people who just want to browse the code might hit the same problem.

Reply:
@jcommelin This is always going to be a problem with using unicode. We have no control over this. The font choices in the GitHub style for this are Menlo, Consolas, Liberation Mono, and Courier; I don't have Menlo but the other three all display something bracket-like here.

Reply:
Ok, I'll install some more fonts

Reply:
Thank you for these helpful remarks.

@urkud I'm not sure; I'm inclined to think it does not need to be localized but I don't have strong feelings, and will happily to defer to anyone who feels differently.

@jcommelin I worried about missing glyphs in other users' font libraries, especially as I having to install extra fonts myself a few weeks ago when I started to look at Mathlib. These little frictions add up. For what it's worth, here are some other candidates:
    * square with underbar:       ⦋ , ⦌  <--- this is what I've chosen for now
    * square with tick at bottom: ⦏ , ⦎
    * square with tick at top:    ⦍ , ⦐
    * square with quill:          ⁅ , ⁆
    * light left tortoise shell:  ❲ , ❳
    * white square:               〚 , 〛
    * full width square:         ［ , ］
    * mathematical white:         ⟦, ⟧ <--- Already in use in Mathlib
I'm happy to switch to any of these if they work for you without installing new fonts.

Reply:
The following currently work for me without new fonts:

* square with quill
* light left tortoise shell
* full width square
* mathematical white

I think the full width version looks best. But that's just one random opinion :wink:

Reply:
I think I missed that you use a Unicode character. No more objections to a global notation.